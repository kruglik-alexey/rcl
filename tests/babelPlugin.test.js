import * as babel from 'babel-core';
import plugin from '../rcl/babelPlugin';

const classTestCase = `
export default class Root extends React.Component {
  render() {
      return React.createElement(
          'div',
          null,
          React.createElement(
              'button',
              { onClick: () => this.setState({}) },
              'Hi'
          ),
          React.createElement(Class, { prop: 'Class' }),
          React.createElement(PureClass, { prop: 'PureClass' }),
          React.createElement(Func, null),
          React.createElement(Func, { prop: 'Func1' })
      );
  }
}
`;

const funcTestCase = `
function func() {
  return React.createElement(
    'div',
    null,
    React.createElement(
        'button',
        { onClick: () => this.setState({}) },
        'Hi'
    ),
    React.createElement(Class, { prop: 'Class' }),
    React.createElement(PureClass, { prop: 'PureClass' }),
    React.createElement(Func, null),
    React.createElement(Func, { prop: 'Func1' })
  );
}
`;

it('classTestCase', () => {
  const {code} = babel.transform(classTestCase, {plugins: [plugin]});
  expect(code).toMatchSnapshot();
});

it('funcTestCase', () => {
  const {code} = babel.transform(classTestCase, {plugins: [plugin]});
  expect(code).toMatchSnapshot();
});
