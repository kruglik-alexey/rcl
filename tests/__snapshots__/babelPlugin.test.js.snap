// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`classTestCase 1`] = `
"
export default class Root extends React.Component {
    render() {
        return React.createElement('div', null, React.createElement('button', { onClick: () => this.setState({}) }, 'Hi'), React.__createElement(1, Class, { prop: 'Class' }), React.__createElement(2, PureClass, { prop: 'PureClass' }), React.__createElement(3, Func, null), React.__createElement(4, Func, { prop: 'Func1' }));
    }
}"
`;

exports[`funcTestCase 1`] = `
"
export default class Root extends React.Component {
    render() {
        return React.createElement('div', null, React.createElement('button', { onClick: () => this.setState({}) }, 'Hi'), React.__createElement(5, Class, { prop: 'Class' }), React.__createElement(6, PureClass, { prop: 'PureClass' }), React.__createElement(7, Func, null), React.__createElement(8, Func, { prop: 'Func1' }));
    }
}"
`;
